tests/bluetooth/tester/src/btp_bap.c

```diff
@@ -819,16 +853,95 @@ static uint8_t bap_discover(const void *cmd, uint16_t cmd_len,
 	return BTP_STATUS_SUCCESS;
 }
 
+static void audio_clock_timeout(struct k_work *work)
+{
+	struct audio_stream *stream;
+	struct k_work_delayable *dwork;
+
+	dwork = k_work_delayable_from_work(work);
+	stream = CONTAINER_OF(dwork, struct audio_stream, audio_clock_work);
+	atomic_inc(&stream->seq_num);
+
+	k_work_schedule(dwork, K_USEC(stream->stream.qos->interval));
+}
+
+static void audio_send_timeout(struct k_work *work)
+{
+	struct bt_iso_tx_info info;
+	struct audio_stream *stream;
+	struct k_work_delayable *dwork;
+	struct bt_iso_chan *iso_chan;
+	struct net_buf *buf;
+	uint32_t size;
+	uint8_t *data;
+	int err;
+	static uint16_t seq_num = 1;
```

Comment:
Can you elaborate on why we have a `static` `seq_num` in this function, rather than just always getting it from `atomic_get(&stream->seq_num);`?

Reply:
I would like to compare here
https://github.com/zephyrproject-rtos/zephyr/blob/1e84b740f9a5c6e04706ac9abf4ce3df5744eac9/tests/bluetooth/tester/src/btp_bap.c#L888
the last accepted sequence number, loaded into `info.seq_num`, with the last requested sequence number. Since the `stream->seq_num` is incremented in a different thread and the `bt_iso_chan_get_tx_sync` can take some time, I need to use `atomic_get(&stream->seq_num)` just before calling the `bt_bap_stream_send`.
At first, I thought that a static variable in a function could hold the last requested sequence number for me. However, I forgot that there will be more streams, so the placement is wrong. Therefore, I will move it into a `last_req_seq_num` field of the `struct audio_stream`.

Reply:
Ah, yes, that makes sense. Perhaps we could store the `last_sen_num` in the `stream` object as well, if we ever want to expand this to support more than 1 stream?